R/data-set.R

```diff
@@ -0,0 +1,80 @@
+#' @title DataSet
+#' @docType class
+#' @description  A wrapper around DataRepository exposing convenience methods to use and manipulate dataSets
+#' (typically observed data) containing an X column, a Y column and potentially an Error columns
+#' @format NULL
+DataSet <- R6::R6Class(
+  "DataSet",
+  inherit = Printable,
+  cloneable = FALSE,
+  active = list(
+    #' @field name The name of the DataSet
+    name = function(value) {
+      if (missing(value)) {
+        return(self$dataRepository$name)
+      }
+      self$dataRepository$name <- value
+    },
+    #' @field xUnit Unit in which the xValues are defined
+    xUnit = function(value){
+      if (missing(value)) {
+        return(private$.xValues$displayUnit)
+      }
+    },
+    #' @field yUnit Unit in which the yValues are defined
+    yUnit = function(value){
+      if (missing(value)) {
+        return(private$.yValues$displayUnit)
+      }
+    },
+    #' @field xValues Values stored in the xUnit dimension (not necessarily in the base unit of the dimension)
+    xValues = function(value){
+      if (missing(value)) {
+        return ( self$dataRepository$baseGrid$values)
+      }
+    }
+  ),
+  public = list(
+    dataRepository = NULL,

```

Comment:
Public for now so that I can debug stuff. It will be private once the implementation is stable